mcodingbot/utils/peps.py

```diff
@@ -48,13 +49,43 @@ async def fetch_pep_info(self, bot: Bot) -> None:
     def get(self, pep_number: int) -> PEPInfo | None:
         return self._peps.get(pep_number)
 
+    def _get_matches_digits(
+        self, query: str, limit: int | None
+    ) -> tuple[Iterable[PEPInfo], int]:
+        """
+        Returns a tuple of (Items found, Amount of items found).
+        """
+        items_iter = (
+            value
+            for key, value in self._peps.items()
+            if str(key).startswith(query)
+        )
+        items = list(itertools.islice(items_iter, limit))
+        return items, len(items)
+
     def search(
         self, query: str, *, limit: int | None = None
     ) -> Iterator[PEPInfo]:
+        yielded = 0
+        if query.isdigit():
+            items, yielded = self._get_matches_digits(query, limit)
+            yield from items
+
+        if limit and yielded >= limit:
+            return
+
         res = fuzzy_search(query, self._pep_map, limit=limit)
+
         for pep in res:
-            if pep_info := self.get(pep[2]):
-                yield pep_info
+            pep_info = self.get(pep[2])
+            if not pep_info:
+                return
+            if limit and yielded >= limit:
+                return
+            if pep_info in items:
+                continue
+            yielded += 1
+            yield pep_info
```

Comment:
I don't feel like this should fail if somehow one of the peps no longer exists in the dict. honestly this could raise an error, but I think a continue makes more sense (and maybe a _LOG.warn)

```py
        for pep in res:
            pep_info = self.get(pep[2])
            if not pep_info or pep_info in items:
                continue

            yield pep_info
            yielded += 1

            if limit and yielded >= limit:
                return
```

Reply:
done